### ISSUE TITLE
`objc` v0.2.1 causes compiler panic on nightly (OSX)

### ISSUE THREAD

**Issue by crazymykl** [rust-lang/rust#36816]
Version: rustc 1.13.0-nightly (a059cb2f3 2016-09-27)

```
   Compiling objc v0.2.1
     Running `rustc /Users/mmacdonald/.cargo/registry/src/github.com-1ecc6299db9ec823/objc-0.2.1/src/lib.rs --crate-name objc --crate-type lib -g -C metadata=1ba30f3cbcb3c55d -C extra-filename=-1ba30f3cbcb3c55d --out-dir /Users/mmacdonald/workspace/foo99/target/debug/deps --emit=dep-info,link -L dependency=/Users/mmacdonald/workspace/foo99/target/debug/deps --extern malloc_buf=/Users/mmacdonald/workspace/foo99/target/debug/deps/libmalloc_buf-7aa64dffd2b03b4f.rlib --cap-lints allow`
error: internal compiler error: unexpected panic

note: the compiler unexpectedly panicked. this is a bug.

note: we would appreciate a bug report: https://github.com/rust-lang/rust/blob/master/CONTRIBUTING.md#bug-reports

note: run with `RUST_BACKTRACE=1` for a backtrace

thread 'rustc' panicked at 'called `Option::unwrap()` on a `None` value', ../src/libcore/option.rs:323
stack backtrace:
   1:        0x112435288 - std::sys::backtrace::tracing::imp::write::hb03a0f85d9cccd1b
   2:        0x1124429ff - std::panicking::default_hook::{{closure}}::h945d25329ef3fdd6
   3:        0x11243fefd - std::panicking::default_hook::hc98de28314f7e4d1
   4:        0x112440596 - std::panicking::rust_panic_with_hook::hf47d6a84575c96f5
   5:        0x112440434 - std::panicking::begin_panic::heb2e59d774c89ce6
   6:        0x112440352 - std::panicking::begin_panic_fmt::h2640a71eaf059834
   7:        0x1124402b7 - rust_begin_unwind
   8:        0x112480640 - core::panicking::panic_fmt::h8e7400f6438ecb75
   9:        0x112480544 - core::panicking::panic::h61585b9964ba1fa2
  10:        0x10e9fe0b3 - rustc_resolve::build_reduced_graph::BuildReducedGraphVisitor::visit_invoc::hedadaf6263d79e9b
  11:        0x10e9f69a2 - rustc_resolve::macros::<impl syntax::ext::base::Resolver for rustc_resolve::Resolver<'a>>::visit_expansion::ha883ee54add676e7
  12:        0x1120758bb - syntax::ext::expand::MacroExpander::collect_invocations::h60ff810e3c6bc674
  13:        0x112072de4 - syntax::ext::expand::MacroExpander::expand::he6cf8a87132aac52
  14:        0x112071644 - syntax::ext::expand::MacroExpander::expand_crate::hff22186693a45a3d
  15:        0x11207ce3d - syntax::ext::expand::expand_crate::hcce563a310499baa
  16:        0x10e30680b - rustc_driver::driver::phase_2_configure_and_expand::{{closure}}::h258525b2788e1899
  17:        0x10e2be2f4 - rustc_driver::driver::phase_2_configure_and_expand::h7891c7a88a497345
  18:        0x10e2b8cd2 - rustc_driver::driver::compile_input::h8a5ca0f49da8c80a
  19:        0x10e2e2188 - rustc_driver::run_compiler::ha0d125e702c7be63
  20:        0x10e222bb0 - std::panicking::try::do_call::h25f69b2f74f3cecc
  21:        0x112442fba - __rust_maybe_catch_panic
  22:        0x10e242594 - <F as alloc::boxed::FnBox<A>>::call_box::h18252cde177b90c5
  23:        0x11243f1b4 - std::sys::thread::Thread::new::thread_start::h66211a1b34bcffec
  24:     0x7fff8bc6399c - _pthread_body
  25:     0x7fff8bc63919 - _pthread_start

error: Could not compile `objc`.
```

@jseyfried: @nikomatsakis thinks you may have some insight into this...

**Comment by jseyfried** [comment]
This was probably caused by #36601 -- I'll investigate.

**Comment by jseyfried** [comment]
Fixed in #36819.
